src/database/queries.rs

```diff
@@ -59,19 +59,24 @@ pub(crate) async fn missing_blocks(conn: &mut PgConnection) -> Result<Vec<u32>>
     .collect())
 }
 
+/// Find missing blocks from the relational database between numbers `min` and
+/// MAX(block_num). LIMIT result to length `max_block_load`.
 pub(crate) async fn missing_blocks_min_max(
     conn: &mut PgConnection,
     min: u32,
+    max_block_load: u32,
 ) -> Result<HashSet<u32>> {
     Ok(sqlx::query_as::<_, (i32,)>(
         "SELECT generate_series
         FROM (SELECT $1 as a, max(block_num) as z FROM blocks) x, generate_series(a, z)
         WHERE
         NOT EXISTS(SELECT id FROM blocks WHERE block_num = generate_series)
         ORDER BY generate_series ASC
```

Comment:
Not your code, but I think this query is hard to read. 
`SELECT generate_series` – don't use the same name as the function used
`SELECT $1 as a, max(block_num) as z FROM blocks) x, generate_series(a, z)` - don't use single letter variable names

Overall I think it's good practice to use in-line comments to explain what each sub-query does.

Also, do we really need to generate a series here? Isn't it enough to figure out what the interval is and return a [`Range`](https://doc.rust-lang.org/std/ops/struct.Range.html)?

Reply:
I agree, the query itself can be made easier to read (a mistake on my end), ie: 

```sql
SELECT missing_number
FROM (SELECT $1 as min_num, MAX(block_num) AS max_num FROM blocks) minmax, 
    GENERATE_SERIES(min_num, max_num) AS missing_number
WHERE
NOT EXISTS (SELECT id FROM blocks WHERE block_num = missing_number)
ORDER BY missing_number ASC
```

On the interval -- I'm not actually sure how we could accomplish this with Postgres (Postgres can't query things that don't exist) other than not using Postgres and just returning all the block numbers and doing it in rust rather then with a query, or generating the series before doing the query, where the query would return numbers which don't exist from the series we give it. 

I think the Postgres solution is good because it contains the set logic to SQL rather than dealing with it in rust

Reply:
> I think the Postgres solution is good because it contains the set logic to SQL rather than dealing with it in rust

Totally agree.

My idea for the `Range` was that we could perhaps return just two numbers from the query, first and last in the range, instead of a whole series. I was thinking about wrapping it in another `SELECT` with the current query as a subquery, but maybe that doesn't work? The idea was: it's a bit annoying we need a to shuffle a whole `HashSet` around when all we need to know (?) is what the extremities of the set are.

Reply:
> it's a bit annoying we need a to shuffle a whole HashSet around when all we need to know (?) is what the extremities of the set are.

If I understand the query correctly, we are not just getting a straight series of numbers and throwing it into a `HashSet`. Instead we first generate the straight series, and then return the numbers from the series that are not in the blocks table. For extremities of the series we only need the query MAX(block_num) since we already feed in the min.